wremnants/CardTool.py

```diff
@@ -597,6 +598,112 @@ def writeForProcesses(self, syst, processes, label, check_systs=True):
         if syst != self.nominalName:
             self.fillCardWithSyst(syst)
 
+    def setSimultaneousABCD(self):
+        # Having 1 process for fakes, for each bin 3 free floating parameters, 2 normalization for lowMT and highMT and one fakerate between iso and anti iso
+        logger.info(f"Set processes for simultaneous ABCD fit")
+
+        # expected fake contribution
+        hist_fake = sum([group.hists[self.nominalName] if name == "Data" else -1*group.hists[self.nominalName] for name, group in self.datagroups.groups.items()])
+        
+        # setting errors to 0
+        hist_fake.view(flow=True)[...] = np.stack((hist_fake.values(flow=True), np.zeros_like(hist_fake.values(flow=True))), axis=-1)
+
+        if "passIso" not in hist_fake.axes.name or "passMT" not in hist_fake.axes.name:
+            raise RuntimeError('"passIso" and "passMT" expected to be found in histogram, but only have axes ')
+
+        # axes in the correct ordering
+        axes = ["passIso", "passMT"]
+        axes = [*axes, "charge"] if "charge" not in self.project else axes
+        axes += [ax for ax in self.project if ax not in axes]
+
+        if set(hist_fake.axes.name) != set(axes) or hist_fake.axes.name[0] != "passIso" or hist_fake.axes.name[1] != "passMT":
+            logger.debug("Axes in histogram are not the same as required or in a different order than expected, try to project")
+            hist_fake = hist_fake.project(*axes)
+
+        # set the expected values in the signal region
+        hist_fake.view(flow=True)[1,1,...] = sel.fakeHistABCD(hist_fake).view(flow=True)
+        
+        fakename = "Nonprompt"
```

Comment:
self.fakeName. I don't really care if it's called Fake or Nonprompt but we should be consistent

Reply:
Yeah, I'm not a big fan of this special treatment, but ok, probably easier to bypass it than to restructure it